src/runtime/runtime.cs

```diff
@@ -2129,7 +2145,7 @@ internal static void Py_CLEAR(ref IntPtr ob)
         //====================================================================
 
         [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
-        internal static extern IntPtr PyCapsule_New(IntPtr pointer, string name, IntPtr destructor);
+        internal static extern NewReference PyCapsule_New(IntPtr pointer, string name, IntPtr destructor);
```

Comment:
Re: implicit conversions to `IntPtr` If you find it hard to change all the places, that use `PyCapsule_New` to use `NewReference` avoiding its dangerous methods as much as possible, just keep this return type `IntPtr` here, and use `NewReference` at the call site.

Reply:
Not that I find it hard, I got confused by how the `BorrowedReference` gets marshalled. Or more like how it could get marshalled without needing an overload. 
Also, NewReference is implicitly castable to `BorrowedReference` which solves most issues. The naming still bugs me a bit, but if it works, it works.